functions/consumer/xmpp-consumer/src/test/java/org/springframework/cloud/fn/consumer/xmpp/XmppConsumerConfigurationTests.java

```diff
@@ -55,30 +54,19 @@
  *
  * @since 4.0.0
  */
-@SpringBootTest(
-		properties = {
-				"xmpp.factory.user=john",
-				"xmpp.factory.password=secret",
-				"xmpp.factory.host=localhost",
-				"xmpp.factory.service-name=localhost",
-				"xmpp.factory.security-mode=disabled"
-		}
-)
+@SpringBootTest
 @DirtiesContext
-@Testcontainers
-public class XmppConsumerConfigurationTests {
-
-	private static final String XMPP_HOST = "localhost";
-	private static final int XMPP_PORT = 5222;
-	private static final String FROM = "john";
-	private static final String TO = "jane";
-	private static final String TO_PW = "secret";
-	private static final String SERVICE_NAME = "localhost";
-
-	@Container
-	private static final DockerComposeContainer XMPP_CONTAINER =
-			new DockerComposeContainer(new File("src/test/docker-compose/xmpp/docker-compose.yml"))
-					.withExposedService("Openfire", XMPP_PORT);
+public class XmppConsumerConfigurationTests implements XmppTestContainerSupport {
+
+	@DynamicPropertySource
+	static void registerConfigurationProperties(DynamicPropertyRegistry registry) {
+		registry.add("xmpp.factory.user", () -> JOHN_USER);
+		registry.add("xmpp.factory.password", () -> USER_PW);
+		registry.add("xmpp.factory.host", () -> XmppTestContainerSupport.getXmppHost());
+		registry.add("xmpp.factory.port", () -> XmppTestContainerSupport.getXmppMappedPort());
+		registry.add("xmpp.factory.service-name", () -> SERVICE_NAME);
+		registry.add("xmpp.factory.security-mode", () -> "disabled");
```

Comment:
Looks like some of these are really not dynamic. What is wrong with specifying via removed `properties`?

Reply:
Agreed, I had a similar question in original PR - this was just copy over from the PR.